File: src/main/java/com/barcelos/recrutamento/core/service/NivelExperienciaService.java
package com.barcelos.recrutamento.core.service;

import com.barcelos.recrutamento.core.exception.ResourceNotFoundException;
import com.barcelos.recrutamento.core.model.NivelExperiencia;
import com.barcelos.recrutamento.core.port.NivelExperienciaRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
public class NivelExperienciaService {

    private final NivelExperienciaRepository nivelExperienciaRepository;

    public NivelExperienciaService(NivelExperienciaRepository nivelExperienciaRepository) {
        this.nivelExperienciaRepository = nivelExperienciaRepository;
    }

    @Transactional
    public NivelExperiencia criar(UUID organizacaoId, String descricao) {
        var nivel = NivelExperiencia.novo(organizacaoId, descricao);
        return nivelExperienciaRepository.save(nivel);
    }

    @Transactional(readOnly = true)
    public NivelExperiencia buscar(UUID nivelId) {
        return nivelExperienciaRepository.findById(nivelId)
                .orElseThrow(() -> new ResourceNotFoundException("Nível de experiência", nivelId));
    }

    @Transactional(readOnly = true)
    public List<NivelExperiencia> listarPorOrganizacao(UUID organizacaoId) {
        return nivelExperienciaRepository.listByOrganizacao(organizacaoId);
    }

    @Transactional
    public NivelExperiencia atualizar(NivelExperiencia nivelExperiencia, String descricao) {
        var nivelAtualizado = NivelExperiencia.atualizar(nivelExperiencia.getId(), nivelExperiencia.getOrganizacaoId(), descricao);
        return nivelExperienciaRepository.save(nivelAtualizado);
    }

    @Transactional
    public void deletar(UUID nivelId) {
        var nivel = buscar(nivelId);
        nivelExperienciaRepository.deleteById(nivelId);
    }
}

File: frontend/recrutamento/src/services/nivelExperiencia.service.ts
import api from "@/lib/api";

export interface NivelExperiencia {
  id: string;
  organizacaoId: string;
  descricao: string;
  ativo: boolean;
}

export const nivelExperienciaService = {
  async listarPorOrganizacao(organizacaoId: string): Promise<NivelExperiencia[]> {
    const response = await api.get(`/organizacoes/${organizacaoId}/niveis-experiencia`);
    return response.data;
  },

  async buscar(organizacaoId: string, nivelExperienciaId: string): Promise<NivelExperiencia> {
    const response = await api.get(`/organizacoes/${organizacaoId}/niveis-experiencia/${nivelExperienciaId}`);
    return response.data;
  },

  async criar(organizacaoId: string, descricao: string): Promise<NivelExperiencia> {
    const response = await api.post(`/organizacoes/${organizacaoId}/niveis-experiencia`, { descricao });
    return response.data;
  },

  async atualizar(organizacaoId: string, nivelExperienciaId: string, descricao: string): Promise<NivelExperiencia> {
    const response = await api.put(`/organizacoes/${organizacaoId}/niveis-experiencia/${nivelExperienciaId}`, { descricao });
    return response.data;
  },

  async deletar(organizacaoId: string, nivelExperienciaId: string): Promise<void> {
    await api.delete(`/organizacoes/${organizacaoId}/niveis-experiencia/${nivelExperienciaId}`);
  },
};
